### ISSUE TITLE
geolocation queries 

### ISSUE THREAD

**Issue by SogoGolf** [Azure/Azure.iOS#111]
is it possible to make a geolocation query from this iOS api ? i couldn't see query params for it but wanted to check.

for example can this be achieved ? 
```
SELECT f.id 
FROM f 
WHERE ST_DISTANCE(f.location, {'type': 'Point', 'coordinates':[151.2357330322266, -33.77001515278013]}) < 30000 
```

**Comment by ftchirou** [comment]
The API does support spatial indexes but the query you quoted is not yet directly possible in the API. It should be straightforward to implement though. @SogoGolf

**Comment by julianparismorgan** [comment]
I actually found a bug with this that I brought up in the Slack channel, and have a fix for in this commit: https://github.com/julianparismorgan/Azure.iOS/commit/9365baaa2defa7a910ceb848d70a85e2fce9b67d 

```let query = Query.select()
                .from("Anchors")
                .where("ST_DISTANCE(Anchors.location, {'type': 'Point', 'coordinates':[122, 22]})", isLessThan: 10)

            AzureData.query(documentsIn: "Anchors", as: CosmosAnchor.self, inDatabase: "Items", with: query) { }```

The query that the SDK creates to send to the service is:
```"SELECT * FROM Anchors WHERE Anchors.ST_DISTANCE(Anchors.location, {\'type\': \'Point\', \'coordinates\':[-122, 47]}) < 10"```

But a properly formatted query is:
```"SELECT * FROM Anchors WHERE ST_DISTANCE(Anchors.location, {'type': 'Point', 'coordinates':[-122, 47]}) < 10"```

So my fix checks for whether it's a geospatial query, and if so, then doesn't append Anchors before ST_DISTANCE.

![image](https://user-images.githubusercontent.com/1396242/53190045-a8e23980-35bd-11e9-883d-55cfbcf64aea.png)

**Comment by NateRickard** [comment]
Closing as I think this was fixed/support was added in PR #114 

@ftchirou make sure you've factored in the above comment/issue in our implementation and reopen if there's any more work to do on this.
